python/ray/experimental/state/state_manager.py

```diff
@@ -0,0 +1,191 @@
+import inspect
+
+from functools import wraps
+
+import grpc
+import ray
+
+from typing import Dict, List
+
+from ray.core.generated.gcs_service_pb2 import (
+    GetAllActorInfoRequest,
+    GetAllActorInfoReply,
+    GetAllPlacementGroupRequest,
+    GetAllPlacementGroupReply,
+    GetAllNodeInfoRequest,
+    GetAllNodeInfoReply,
+    GetAllWorkerInfoRequest,
+    GetAllWorkerInfoReply,
+)
+from ray.core.generated.node_manager_pb2 import (
+    GetTasksInfoRequest,
+    GetTasksInfoReply,
+    GetNodeStatsRequest,
+    GetNodeStatsReply,
+)
+from ray.core.generated import gcs_service_pb2_grpc
+from ray.core.generated.node_manager_pb2_grpc import NodeManagerServiceStub
+from ray.core.generated.reporter_pb2_grpc import ReporterServiceStub
+from ray.dashboard.modules.job.common import JobInfoStorageClient, JobInfo
+
+
+class StateSourceNetworkException(Exception):
+    """Exceptions raised when there's a network error from data source query."""
+
+    pass
+
+
+def apply_state_client_api_policy(func):
```

Comment:
I find this name pretty vague/generic. Maybe something like `handle_network_errors` is more clear? Can update it if more logic is added...

Reply:
Hmm yeah lol. I like that name better (also given retry will be for mostly network errors as well)